veloroutes_voies_vertes/processing/covadis/export_covadis.py

```diff
@@ -183,21 +184,13 @@ def to_shapefile(self, context, table, dpt, dirname, uri, geom, charger):
         options.fileEncoding = "utf-8"
 
         # Enregistrement du fichier shape
-        error = QgsVectorFileWriter.writeAsVectorFormatV2(
+        QgsVectorFileWriter.writeAsVectorFormatV2(
             layer=layer,
             fileName=dirname + '/' + filename + ".shp",
             transformContext=transformContext,
             options=options
         )
 
-        # Vérification du retour de l'enregistrement et des erreurs potentiels
-        if error[0] == QgsVectorFileWriter.ErrCreateLayer:
-            raise QgsProcessingException(
-                "Erreur lors de l'écriture du fichier :\n"
-                "Vérifiez que le chemin de destination est valide")
-        if error[0] != QgsVectorFileWriter.NoError:
-            raise QgsProcessingException("Erreur lors de l'écriture du fichier :" + error[1])
```

Comment:
@rldhont @Gustry Je propose de retirer ce test car il bloque les fichiers temporaires

Reply:
Avant de supprimer, quel message reçois-tu ?

En quoi ce test bloquerait ?

Reply:
@rldhont On reçoit le message que construit le `raise  QgsProcessingException` la variable error est définie lors de l'écriture dans le fichier ici: https://github.com/3liz/qgis-veloroutes_voies_vertes-plugin/pull/81/files#diff-f175f337a27a11f26227ac2a4919fbdb4ae20702724052b692ef889828bc6846L186-R185
Et si je retire ces tests le fichier temporaire marche.

Reply:
Lequel des 2 messages est envoyé ? Il est tout de même important de savoir si il y a une erreur lors de la création du fichier.

Reply:
> Et si je retire ces tests le fichier temporaire marche.

Le fichier temporaire est bien crée et valide ? (quand tu recharges la nouvelle couche dans QGIS)
Que contient le contenu de `error[1]` qui correspond au contenu l'erreur.

Reply:
@Gustry Oui le fichier est la et il y a les données, le error[1] contient: La création de la couche a échoué (Erreur OGR : Failed to open Shape DBF file `/tmp/processing_e32c388defce43aabbc8abee999657eb/1034788a27be4d93b83ffda32da5e7e6/FOLDER/N_3V_ITINERAIRE_066.dbf'.)

Reply:
@rldhont en fait le premier est renvoyé, et si j'enlève le test le deuxième est renvoyé. Si j'enlève les deux tests alors ça fonctionne

Reply:
> Oui le fichier est la 

@drillinP Tu peux faire un `ls -l` sur le chemin/le dossier ?
Et/ou `ogrinfo /le/chemin/sur/le/shapefile.shp` ?

`ls -l /tmp/processing_e32c388defce43aabbc8abee999657eb/1034788a27be4d93b83ffda32da5e7e6/FOLDER/` par ex

Reply:
ls -l  Aucun fichier ou dossier de ce type @Gustry

Reply:
Et donc il s'enregistre où ce fichier ?

Reply:
Alors je viens de regarder en fait c'est pas le tempfile qu'il me retourne mais la couche donc les tempfile ne marche pas

Reply:
> ls -l Aucun fichier ou dossier de ce type

C'était un indice que cela ne marchait pas ...

Reply:
J'ai trouvé comment régler le problème @Gustry @rldhont. en utilisant `QgsProcessingUtils.generateTempFilename` pour créer le tempfile avec d'exécuter la méthode `QgsVectorFileWriter.writeAsVectorFormatV2` permet de créer le shp temporaire. En revanche il me faut détecter s'il s'agit d'un dossier temporaire car si ce n'en est pas un `QgsProcessingUtils.generateTempFilename` n'est pas nécessaire

Reply:
Pas besoin d'utiliser `QgsProcessingUtils.generateTempFilename` car Processing te donne déjà le nom du fichier temporaire. Il n'est donc pas utile de regénérer un fichier temporaire, utilise celui de Processing.

`QgsVectorFileWriter.writeAsVectorFormatV2` n'arrive pas a écrire le fichier car le dossier n'existe pas. Il faut donc  récupérer le dossier parent de chemin et le créer si il n'existe pas @drillinP